ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -0,0 +1,76 @@
+from typing import List
+
+from mlagents.trainers.buffer import AgentBuffer, BufferException
+
+
+class ProcessingBuffer(dict):
+    """
+    ProcessingBuffer contains a dictionary of AgentBuffer. The AgentBuffers are indexed by agent_id.
+    """
+
+    def __str__(self):
+        return "local_buffers :\n{0}".format(
+            "\n".join(["\tagent {0} :{1}".format(k, str(self[k])) for k in self.keys()])
+        )
+
+    def __getitem__(self, key):
+        if key not in self.keys():
+            self[key] = AgentBuffer()
+        return super().__getitem__(key)
+
+    def reset_local_buffers(self) -> None:
+        """
+        Resets all the local local_buffers
+        """
+        agent_ids = list(self.keys())
```

Comment:
rewrite as 
``` python
for buf in self.values():
    buf.reset_agent()
```
(or something like that) ?  Forming the list is unnecessary (allocates memory) and you're doing key lookups that you don't really need.

Reply:
Done